main.go

```diff
@@ -246,3 +256,52 @@ func bindFlags(cmd *cobra.Command, v *viper.Viper) {
 		}
 	})
 }
+
+// Create an issue in the provider repo with a message describing the upgrade failure
+func createFailureIssue(ctx upgrade.Context, repoOrg string, repoName string, errMsg string) (string, error) {
+	now := time.Now()
+	y, m, d := now.Date()
+	title := fmt.Sprintf("Upgrade provider failure: %v %v %v", y, m, d)
+
+	getIssues := exec.CommandContext(ctx, "gh", "search", "issues",
+		"Upgrade provider failure: ",
+		"--repo="+repoOrg+"/"+repoName,
+		"--json=title,number",
+	)
+	bytes := new(bytes.Buffer)
+	getIssues.Stdout = bytes
+	err := getIssues.Run()
+	if err != nil {
+		return "Failed to create failure issue: failed to search existing issues", err
+	}
+	titles := []struct {
+		Title  string `json:"title"`
+		Number int    `json:"number"`
+	}{}
+	err = json.Unmarshal(bytes.Bytes(), &titles)
+	if err != nil {
+		return "Failed to create failure issue: failed to unmarshal gh output", err
+	}
+	for _, t := range titles {
+		cmd := exec.Command("gh", "issue", "delete",
+			fmt.Sprintf("%v", t.Number),
```

Comment:
I don't think we should delete here; as I think we expect issues to persist.

Comment on the issue instead that a new run failed with a link to the workflow ID?